tensorflow/lite/kernels/internal/reference/add.h

```diff
@@ -138,6 +143,24 @@ inline void Add(const ArithmeticParams& params,
                 const RuntimeShape& output_shape, int16* output_data) {
   TFLITE_DCHECK_LE(params.quantized_activation_min,
                    params.quantized_activation_max);
+  const int flat_size =
+      MatchingElementsSize(input1_shape, input2_shape, output_shape);
+
+  int max_value = std::numeric_limits<int16>::max();
+
+  TFLITE_DCHECK_GT(params.input1_offset, -max_value);
+  TFLITE_DCHECK_GT(params.input2_offset, -max_value);
+  TFLITE_DCHECK_LT(params.input1_offset, max_value);
+  TFLITE_DCHECK_LT(params.input2_offset, max_value);
+  AddElementwise(flat_size, params, input1_data, input2_data, output_data);
+}
+
+inline void AddLSTM(const ArithmeticParams& params,
```

Comment:
What is AddLSTM?

Reply:
The present version needs scales to be POT as they are for LSTM. This PR extends this version to include general scales as for 8-bit. I renamed the current version to AddLSTM to indicate that this function is specific to LSTM.

Reply:
I am not sure I understand, this Add and Sub operation are specific to LSTM? 

How are they created in the TFLite graph. If possible we should avoid having unfused versions of add and sub that are specific to LSTM.

Reply:
No, the existed function was named Add before, but it was only with POT scale. We needed a general version, so I added the implementation similar to 8-bit and renamed the existed function to AddLSTM to indicate that this function is used when scale is POT.
This is probably a bad name.

Reply:
Ok sorry for the delay, was figuring out what the right thing to do here was.

This is unfortunate that there was an exisiting version of the 16bit code was specific to POT.

We have to take extra care for versioning in this case. 

Here is what i propose:
1) Add a new attribute for this operation called pot_scale. This attribute must default to True.

2) Add your new code path under the case where pot_scale=false.

3) when we use this operation in the converter and add the new op version, we will set the attribute to False. 

Does that seem reasonable, and make sense?